daemon/cdi_linux.go

```diff
@@ -0,0 +1,24 @@
+package daemon
+
+import (
+	"github.com/docker/docker/pkg/capabilities"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// registerCDIDriver installs a device driver for CDI devices.
+// The driver injects CDI devices into an incoming OCI spec and is called for DeviceRequests associated with CDI devices.
+func registerCDIDriver() error {
+	driver := &deviceDriver{
+		capset:     capabilities.Set{"cdi": struct{}{}},
+		updateSpec: injectCDIDevices,
+	}
+
+	registerDeviceDriver("cdi", driver)
+
+	return nil
+}
+
+// injectCDIDevices is an updateSpec hander for the CDI device driver
+func injectCDIDevices(s *specs.Spec, dev *deviceInstance) error {
+	return injectDevices(s, dev.req.DeviceIDs)
+}
```

Comment:
This is currently required because `deviceInstance` is a linux-specific type. This indirection can be removed one we have refactored the driver registration as previously discussed.

Reply:
So move `injectDevices` into `cdi_linux.go`, then. Make the trivial wrapper go away.